src/main/java/sirius/db/util/AutoClosingStream.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.db.util;
+
+import sirius.kernel.commons.Explain;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.BinaryOperator;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.IntFunction;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.function.ToDoubleFunction;
+import java.util.function.ToIntFunction;
+import java.util.function.ToLongFunction;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+/**
+ * A stream wrapper calling {@link #close()} after any terminal operation.
+ *
+ * @param <T> the type of the stream elements
+ */
+public record AutoClosingStream<T>(Stream<T> delegate) implements Stream<T> {
```

Comment:
whats the benefit of making this a record?

Reply:
There is none, its just recommended by IntelliJ

There is one tiny detail you cannot achieve without a record, which is that the `delegate` is truely constant this way. If you "just" declare it `final`, you can still modify it via Reflection. But I would not consider this a real "benefit".